modules/imap/hm-imap.php

```diff
@@ -1090,7 +1090,13 @@ public function search($target='ALL', $uids=false, $terms=array(), $esearch=arra
             if ($esearch_enabled) {
                 $res = $esearch_res;
             }
-            return $this->cache_return_val($res, $cache_command);
+            $found_messages_uids = $this->cache_return_val($res, $cache_command);
```

Comment:
Are we sure that we get all search results here? If they are paginated, sorting won't work afterwards.
Also, isn't there an IMAP command to search AND sort in one go?

Reply:
> Are we sure that we get all search results here? If they are paginated, sorting won't work afterwards. Also, isn't there an IMAP command to search AND sort in one go?

According to my researches, yes: we get all of them, since the IMAP response (`$status`) is "OK" there.
Pagination is being processed when formatting as I observed.

Search and sort in one go with IMAP command: I'll look into this possibility and share my findings, so that you can advise.